### PR TITLE
Fix HTTPSender ignoring query parameters

### LINKED ISSUE #301: jaeger-collector returning 'Unsupported format type'
## Requirement - what kind of business use case are you trying to solve?

Trying to use jaeger-client-node together with istio and ambassador

## Problem - what in Jaeger blocks you from solving the requirement?

Jaeger is not respecting the defined `format` in HTTPSender.
Right now I use `collectorEndpoint: 'http://jaeger-collector.istio-system:14268/api/traces?format=jaeger.thrift',` and the `?format=jaeger.thrift` is not sent with any of the flushes.
Further more it's quite hard to debug, as there's no errors when Jaeger responds with `Unsupported format type:`, it's just silently ignored.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

I propose to change https://github.com/jaegertracing/jaeger-client-node/blob/206e4c1c4562c0ccbf6e4f3617da5d72baf9bb39/src/reporters/http_sender.js#L66
Instead of path: `this._url.pathname,` I believe it should be `path: this._url.path,`.

I've tried the fix above locally, and it fixes my issues.

### PR DESCRIPTION
**Pull request by Multiply**
## Which problem is this PR solving?

Resolves #301 

## Short description of the changes

Changed HTTPSender to include query parameters from `collectorEndpoint`.

### CODE REVIEW

**Comment by Multiply** [comment]
@yurishkuro I might be running an older version of jaeger-collector, in that case. :(

File: src/reporters/http_sender.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ export default class HTTPSender {
       protocol: this._url.protocol,
       hostname: this._url.hostname,
       port: this._url.port,
-      path: this._url.pathname,
+      path: this._url.path,
```
Comment (yurishkuro) [inline src/reporters/http_sender.js]: I am not seeing `path` being a valid property of the URL object: https://nodejs.org/api/url.html#url_class_url

**Comment by yurishkuro** [comment]
Do you have problems with upgrading?

**Comment by Multiply** [comment]
@yurishkuro I am investigating how big a deal it is. It's currently running in 3 docker images that we're not in control of, so I might have to wait for those to be upgraded to latest jaeger.

File: src/reporters/http_sender.js (hunk 1 above)
Comment (Multiply) [reply src/reporters/http_sender.js]: https://nodejs.org/api/url.html#url_urlobject_path

**Comment by Multiply** [comment]
Closing this, as I don't see a point in supporting older versions of jaeger.

**Pull request closed without merging**